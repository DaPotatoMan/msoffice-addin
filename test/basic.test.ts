import { fileURLToPath } from 'node:url'
import { describe, expect, it } from 'vitest'
import { $fetch, setup } from '@nuxt/test-utils/e2e'

describe('ssr', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('../playground', import.meta.url)),
  })

  it('renders office manifest', () => {
    expect($fetch('/office-app/manifest.xml'))
      .resolves.toMatchSnapshot()
  })

  it('injects office.js code', () => {
    const SCRIPT_URL = 'https://appsforoffice.microsoft.com/lib/1.1/hosted/office.js'

    expect($fetch('/'))
      .resolves
      .toContain(SCRIPT_URL)

    expect($fetch('/404.html'))
      .resolves
      .not.toContain(SCRIPT_URL)
  })
})
