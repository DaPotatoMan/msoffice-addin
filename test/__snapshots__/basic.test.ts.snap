// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ssr > renders office manifest 1`] = `
"<?xml version="1.0" encoding="UTF-8"?>
<OfficeApp xmlns="http://schemas.microsoft.com/office/appforoffice/1.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="TaskPaneApp">
  <Id>c6890c26-5bbb-40ed-a321-37f07909a2f0</Id>
  <Version>1.0</Version>
  <ProviderName>Contoso, Ltd</ProviderName>
  <DefaultLocale>en-US</DefaultLocale>
  <DisplayName DefaultValue="Contoso App" />
  <Description DefaultValue="Office add-in for Contoso App"/>
  <SupportUrl DefaultValue="https://contoso.com/support" />
  <AppDomains>
    <AppDomain>https://contoso.com</AppDomain>
  </AppDomains>
  <DefaultSettings>
    <SourceLocation DefaultValue="https://www.contoso.com/search_app/Default.aspx" />
  </DefaultSettings>
  <Permissions>ReadWriteDocument</Permissions>
  <WebApplicationInfo>
    <Id>c6890c26-5bbb-40ed-a321-37f07909a2f0</Id>
    <Resource>api://contoso.com/outlook/c6890c26-5bbb-40ed-a321-37f07909a2f0</Resource>
    <Scopes>
      <Scope>openid</Scope>
      <Scope>profile</Scope>
      <Scope>User.Read</Scope>
    </Scopes>
  </WebApplicationInfo>
</OfficeApp>"
`;
